optax/_src/transform.py

```diff
@@ -869,6 +869,64 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByNewtonState(NamedTuple):
+  """State holding the inverse of the sum of gradient squares to date."""
+
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u, v):
```

Comment:
Please add type annotations to this function (even if we decide to make it private in the end).